compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3553,6 +3555,70 @@ TR::Register *OMR::Power::TreeEvaluator::vmulInt32Helper(TR::Node *node, TR::Cod
    return TR::TreeEvaluator::inlineVectorBinaryOp(node, cg, TR::InstOpCode::vmuluwm);
    }
 
+TR::Register *OMR::Power::TreeEvaluator::vmulInt64Helper(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   if (cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P10))
+      return TR::TreeEvaluator::inlineVectorBinaryOp(node, cg, TR::InstOpCode::vmulld);
+
+   /*
+   To perform the multiplication without the vmulld instruction, we will break it into parts.
+
+   Suppose we have two long integers (64 bits), AB and CD, where each of A, B, C, and D is one word (32 bits).
+   To get their product, AB * CD, we can do the following:
+
+   AB * CD = B*D + 2^32*(AD + BC) + 2^64*(A*C)
```

Comment:
This is a bit confusing, since `AB` and `CD` seem to represent bit concatenations whereas `AD` and `BC` seem to be used here (and in a couple other locations in the comments) to represent multiplications. You should probably use `A*D` and `B*C` instead.